insights/parsers/losetup.py

```diff
@@ -0,0 +1,83 @@
+"""
+LoSetup - command ``/usr/sbin/losetup -l``
+==========================================
+
+This parser reads the output of ``/usr/sbin/losetup -l`` into a list of entries.
+Each entry is a dictionary of headers:
+
+* ``NAME`` - the path name of the loop back device (strings)
+* ``SIZELIMIT`` - the data end position of backing file in bytes (integer)
+* ``OFFSET`` - the data start position of backing file in bytes (integer)
+* ``AUTOCLEAR`` - the autoclear flag (boolean)
+* ``RO`` - the read only flag (boolean)
+* ``BACK-FILE`` - the path of the backing file (strings)
+* ``DIO`` - the direct I/O flag (boolean)
+* ``LOG-SEC`` - the logical sector size of the loop device in bytes (integer)
+
+Sample output of ``losetup -l`` command is::
+
+    NAME       SIZELIMIT OFFSET AUTOCLEAR RO BACK-FILE      DIO LOG-SEC
+    /dev/loop0         0      0         0  0 /root/disk.img   1     512
+
+Examples:
+
+    >>> type(losetup)
+    <class 'insights.parsers.losetup.LoSetup'>
+    >>> len(losetup)
+    1
+    >>> losetup[0]['NAME']
+    '/dev/loop0'
+    >>> losetup[0]['RO']
+    False
+    >>> losetup[0]['DIO']
+    True
+    >>> losetup[0]['NAME']
+    '/dev/loop0'
+    >>> losetup[0]['SIZELIMIT']
+    0
+    >>> losetup[0]['OFFSET']
+    0
+    >>> losetup[0]['AUTOCLEAR']
+    False
+    >>> losetup[0]['RO']
+    False
+    >>> losetup[0]['BACK-FILE']
+    '/root/disk.img'
+    >>> losetup [0]['DIO']
+    True
+    >>> losetup[0]['LOG-SEC']
+    512
+"""
+from insights import CommandParser, parser
+from insights.specs import Specs
+from insights.parsers import parse_delimited_table, SkipException
+
+
+@parser(Specs.losetup)
+class LoSetup(CommandParser, list):
+    """
+    Parses the output of the ``/usr/sbin/losetup -l`` command.
+
+    Attributes:
+        data (list): a list of parsed output of the command. Each entry is a dictionary of headers.
+    """
+
+    def parse_content(self, content):
+        if not content:
+            raise SkipException("Empty output.")
+
+        self.extend(parse_delimited_table(content))
+        for entry in self.data:
+            for key in ['SIZELIMIT', 'OFFSET', 'LOG-SEC']:
+                if key in entry:
+                    entry[key] = int(entry[key])
+            for key in ['AUTOCLEAR', 'RO', 'DIO']:
+                if key in entry:
+                    entry[key] = True if entry[key] == '1' else False
+
+    @property
+    def data(self):
+        """
+        Set data as property to keep compatibility
```

Comment:
@takayuki-nagata - Sorry for the late comment.

Generally, this `self.data` attribute is not required for the newly developed parser, it's recommended for users to use it as `a` `list` parser directly. so, it's not necessary to add this `property` to keep backward compatibility. Would you please remove it here and from the above `Attributes` section in the doc part? Thanks.

Reply:
@xiangce sure, I have removed the property and have re-sent v3.